### ISSUE TITLE
should there be a stringifier and what should it return?

### ISSUE THREAD

**Issue by wanderview** [WICG/urlpattern#47]
Should there be a stringifier for URLPattern?  If so, what should it return?

There is no single string input to URLPattern that works for all patterns, so we can't use that as the toString() output.

We could add a general human readable string, but its unclear if that is necessary if devtools already supports iterating properties.  With per-component pattern properties its already quite debuggable:

![image](https://user-images.githubusercontent.com/1256526/110354172-ddd75280-8005-11eb-9fc7-aa4e6f735296.png)

**Comment by wanderview** [comment]
I guess if we add per-component regex getters as well this devtools view will get a bit busier.  Having a minimal view of the pattern might be useful.  But I will probably wait to see what developers say here.

**Comment by wanderview** [comment]
To clarify, @domenic suggested not providing a stringifier since its likely developers will try to use it as a serialization mechanism and we can't safely produce a single string that can be deserialized with the constructor.  We could provide a toJSON() method if folks need that, though.

**Comment by wanderview** [comment]
Now that there is a constructor that takes a single string as an argument, should we revisit whether to provide a stringifier?  @domenic, what do you think?

**Comment by domenic** [comment]
Yeah if you can guarantee that `new URLPattern(urlPattern.toString())` behaves the same as `urlPattern` then a stringifier makes a lot of sense. That's not a trivial property, but it seems doable...

**Comment by wanderview** [comment]
After thinking about this some more I think there are URLPattern objects that can be created with URLPatternInit dictionaries that we still cannot parse from a constructor string.  Therefore I think leaving this without a stringifier for now is the right decision.

This is captured by the spec, so closing this.
